selfdrive/car/docs_definitions.py

```diff
@@ -31,60 +30,68 @@ class Star(Enum):
-  hyundai_o = "Hyundai O"
-  hyundai_p = "Hyundai P"
-  hyundai_q = "Hyundai Q"
-  custom = "Developer"
-  obd_ii = "OBD-II"
-  gm = "GM"
-  nissan_a = "Nissan A"
-  nissan_b = "Nissan B"
-  mazda = "Mazda"
-  ford_q3 = "Ford Q3"
-  ford_q4 = "Ford Q4"
-  none = "None"
-
-
-class HarnessPart(Enum):
+class CarPart(Enum):
+  nidec_connector = "Honda Nidec connector"
+  bosch_a_connector = "Honda Bosch A connector"
+  bosch_b_connector = "Honda Bosch B connector"
+  toyota_connector = "Toyota connector"
+  subaru_a_connector = "Subaru A connector"
+  subaru_b_connector = "Subaru B connector"
+  fca_connector = "FCA connector"
+  ram_connector = "Ram connector"
+  vw_connector = "VW connector"
+  j533_connector = "J533 connector"
+  hyundai_a_connector = "Hyundai A connector"
+  hyundai_b_connector = "Hyundai B connector"
+  hyundai_c_connector = "Hyundai C connector"
+  hyundai_d_connector = "Hyundai D connector"
+  hyundai_e_connector = "Hyundai E connector"
+  hyundai_f_connector = "Hyundai F connector"
+  hyundai_g_connector = "Hyundai G connector"
+  hyundai_h_connector = "Hyundai H connector"
+  hyundai_i_connector = "Hyundai I connector"
+  hyundai_j_connector = "Hyundai J connector"
+  hyundai_k_connector = "Hyundai K connector"
+  hyundai_l_connector = "Hyundai L connector"
+  hyundai_m_connector = "Hyundai M connector"
+  hyundai_n_connector = "Hyundai N connector"
+  hyundai_o_connector = "Hyundai O connector"
+  hyundai_p_connector = "Hyundai P connector"
+  hyundai_q_connector = "Hyundai Q connector"
+  custom_connector = "Developer connector"
+  obd_ii_connector = "OBD-II connector"
+  gm_connector = "GM connector"
+  nissan_a_connector = "Nissan A connector"
+  nissan_b_connector = "Nissan B connector"
+  mazda_connector = "Mazda connector"
+  ford_q3_connector = "Ford Q3 connector"
+  ford_q4_connector = "Ford Q4 connector"
+  none_connector = "None connector"
   harness_box = "harness box"
   comma_power_v2 = "comma power v2"
-  rj45_cable = "RJ45 cable (7 ft)"
+  rj45_cable_7ft = "RJ45 cable (7 ft)"
   long_obdc_cable = "long OBD-C cable"
   usbc_coupler = "USB-C coupler"
+  comma_3 = "comma 3"
+  mount = "mount"
+  angled_mount = "angled mount"
+  red_panda = "red panda"
+  usb_a_2_a_cable = "USB A-A cable"
+  usbc_otg_cable = "USB C OTG cable"
+  obd_c_cable_1point5ft = "OBD-C cable (1.5 ft)"
 
 
-DEFAULT_HARNESS_PARTS: List[HarnessPart] = [HarnessPart.harness_box, HarnessPart.comma_power_v2, HarnessPart.rj45_cable]
+DEFAULT_CAR_PARTS: List[CarPart] = [CarPart.harness_box, CarPart.comma_power_v2, CarPart.rj45_cable_7ft]
 
 
 @dataclass
-class HarnessKit:
-  connector: Harness = Harness.none
-  parts: List[HarnessPart] = field(default_factory=lambda: copy.copy(DEFAULT_HARNESS_PARTS))
+class CarParts:
+  def __init__(self, parts: List[CarPart] = None):
+    self.parts = parts or []
+
+  @classmethod
+  def default(cls, parts: List[CarPart], default: List[CarPart] = None):
+    return cls(parts + (default or DEFAULT_CAR_PARTS))
```

Comment:
either here or as a test, we should make sure that a connector part exists for all cars with a parts list